### DISCUSSION TITLE
How can I force full reload after redirect?

### DISCUSSION THREAD

**Discussion by kucharzyk** [unpoly/unpoly#359] [Q&A] [answered]
Hello,

I'm using unpoly to render only part of page on navigation. In case of authenticated page and expired jwt I'm sending 303 (see other) redirect to navigate user to the login screen but in this case I would like to refresh entire body not only default target. For example when I am submiting form only this form should be refreshed but if token expires unpoly refresh form window and put login page there. I've tried to add Up-target header while redirecting but it is not working. I think that unpoly gets only headers after redirect. Any thoughts how can I fix it?

**Comment by adam12** [comment]
I'd try sending a special header along with the 303, and then checking for the header once Unpoly finishes the request. If the header is present, call `navigate()` on the request.

This is a snippet from a 1.0 project of mine, but perhaps you can modify it to suit Unpoly 2.

```javascript
up.on(document, "up:proxy:loaded", (ev) => {
  if (ev.response.getHeader("x-up-navigate") === "true") {
    ev.preventDefault()
    up.history.config.enabled = false
    /* To prevent flash of unstyled body, clear it on next tick */
    up.util.task(() => {
      document.body.outerHTML = null
      return ev.request.navigate()
    })
  }
})
```

**Comment by kucharzyk** [comment]
Thanks @adam12

I've tried something similiar with Unpoly2

```js
up.on('up:fragment:loaded', (event) => {
  let isMaintenancePage = event.response.getHeader('X-Maintenance')

  if (isMaintenancePage) {
    // Prevent the fragment update and don't update browser history
    event.preventDefault()

    // Make a full page load for the same request.
    event.request.loadPage()
  }
})
```

It's example from documentation (https://unpoly.com/up:fragment:loaded)

Unfortunately headers from original response are not accesible here.
Only requests from redirected page.

**Comment by kucharzyk** [comment, marked as answer]
I temporary fixed it by using X-Up-Target: body header on login page. 

Is there any better solution?
